gettsim/interface.py

```diff
@@ -531,6 +538,78 @@ def _root_nodes(dag):
             yield node
 
 
+def _add_rounding_for_one_function(base, direction):
+    """Decorator to round the output of a function.
+
+    Parameters
+    ----------
+    base : float
+        Precision of rounding (e.g. 0.1 to round to the first decimal place)
+    round_d : bool
+        Whether rounding should be applied
+    direction : str
+        Whether the series should be rounded up, down or to the nearest number
+
+    Returns
+    -------
+    results : pandas.Series
+        Series with (potentially) rounded numbers
+    """
+
+    def inner(func):
+        # Make sure that signature is preserved
+        @functools.wraps(func)
+        def wrapper(*args, **kwargs):
+            out = func(*args, **kwargs)
+            if direction == "up":
+                return base * np.ceil(out / base)
+            elif direction == "down":
+                return base * np.floor(out / base)
+            elif direction == "nearest":
+                return base * (out / base).round()
+            else:
+                raise ValueError("direction must be one of 'up', 'down', or 'nearest'")
+
+        return wrapper
+
+    return inner
+
+
+def _add_rounding_to_functions(functions):
+    """Add appropriate rounding of outputs to functions
+
+    Parameters
+    ----------
+    functions : dict of callable
+        Dictionary of functions which are either internal or user provided functions.
+
+
+    Returns
+    -------
+    functions_new : dict of callable
+        Dictionary mapping function names to callables for whose outputs appropriate
+        rounding is applied.
+
+    """
+    functions_new = {}
+    for f_name, func in functions.items():
+        if hasattr(func, "__roundingspec__"):
+            try:
```

Comment:
Instead of a try-except block I would just check that all required keys are there and raise a ValueError if not. 

If you stick to the try-except, you should handle `SystemExit` and `KeyboardInterrupt` separately. Here is an [example](https://github.com/OpenSourceEconomics/estimagic/blob/1507ed18e68958cb3278ed97e224a6abd189b201/src/estimagic/optimization/internal_criterion_template.py#L158).